### DISCUSSION TITLE
Close : Search for duplicates in documents.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#223] [Q&A] [answered]
I want to have a query that gives me the duplicates by displaying each line concerned by the duplicate

with my query BiblioteQ shows me the duplicate title but not each record - is this possible?

SELECT COUNT(*) AS nbr_doublon, accession_number, title FROM book GROUP BY title HAVING COUNT(*) > 1

**Comment by textbrowser** [comment]
I need to know the criteria of a duplicate.

**Reply by meteos77** [reply]
in my test database

accession_number 1 and 2 have the same title
accession_number 6 is shared by 3 documents title 6 7 and 8

[Test-Doublons.sqlite.tar.gz](https://github.com/textbrowser/biblioteq/files/9228142/Test-Doublons.sqlite.tar.gz)

I think BiblioteQ protects us from having duplicate isbns but my old software allowed it
2 identical isbn could cohabit with 2 different accession numbers

**Comment by meteos77** [comment]
currently if I do the query on my entire imported database 11175 documents

SELECT COUNT() AS nbr_double, * FROM book GROUP BY title HAVING COUNT() > 1
- I get 799 duplicate documents on the title

SELECT COUNT() AS nbr_doublon, * FROM book GROUP BY accession_number HAVING COUNT() > 1
- I get 279 duplicate documents on accession_number

**Reply by meteos77** [reply]
so I have some cleaning to do :-)

**Comment by meteos77** [comment]
SQL is magic after having found a good site with the right example, I think I have my answer :-)

SELECT a.* FROM book AS a JOIN (SELECT accession_number, COUNT(*) FROM book GROUP BY accession_number HAVING COUNT(*) > 1 ) AS b ON a.accession_number = b.accession_number

SELECT a.* FROM book AS a JOIN (SELECT title, COUNT() FROM book GROUP BY title HAVING COUNT() > 1 ) AS b ON a.title =  b.title

**Comment by textbrowser** [comment, marked as answer]
I'm confused.

OK, finding all entries which have duplicate titles:

> SELECT accession_number, myoid FROM book WHERE title IN (SELECT title FROM book GROUP BY title HAVING COUNT(title) > 1);

Finding all entries with duplicate accession numbers:

> SELECT accession_number, myoid FROM book WHERE accession_number IN (SELECT accession_number FROM book GROUP BY accession_number HAVING COUNT(accession_number) > 1);

**Comment by meteos77** [comment]
do you validate my requests?

**Reply by textbrowser** [reply]
Validate your requests?

**Reply by meteos77** [reply]
Validate sql queries

**Reply by textbrowser** [reply]
> Validate sql queries

No. SQLite will do that.

**Reply by textbrowser** [reply]
And you can look at the results of your queries to determine if they produce what you're looking for.

**Reply by meteos77** [reply]
I'm not going to count the 11000 lines of my database to see if the request has no defect, I ask the specialist :-)

**Reply by meteos77** [reply]
your opinion was requested for educational purposes

**Reply by meteos77** [reply]
I hope you have a good day on Sunday

**Reply by textbrowser** [reply]
I see. If there is much data, you can curtail it with LIMIT and OFFSET. :)

**Reply by textbrowser** [reply]
Enjoy the remainder of Sunday in France.

**Comment by textbrowser** [comment]
Sometimes I don't understand your questions. At the beginning of the discussion, you didn't include the criteria for uniqueness. Then you kind of described it. I wish the requests to be simple.

"I want to find all books which have identical titles."

**Reply by meteos77** [reply]
sorry: I will try to be more clear in my next request
